internal/stores/metadata.go

```diff
@@ -419,22 +419,32 @@ func (s *SQLStore) Object(ctx context.Context, key string) (object.Object, error
 }
 
 func (db *SQLStore) RecordContractSpending(ctx context.Context, records []api.ContractSpendingRecord) error {
+	squashedRecords := make(map[types.FileContractID]api.ContractSpending)
 	for _, r := range records {
+		squashedRecords[r.ContractID] = squashedRecords[r.ContractID].Add(r.ContractSpending)
+	}
+	for fcid, cs := range squashedRecords {
 		err := db.db.Transaction(func(tx *gorm.DB) error {
 			var c dbContract
 			err := tx.Model(&dbContract{}).
-				Where("fcid = ?", fileContractID(r.ContractID)).
+				Where("fcid = ?", fileContractID(fcid)).
 				Take(&c).Error
 			if errors.Is(err, gorm.ErrRecordNotFound) {
 				return nil // contract not found, continue with next one
```

Comment:
should we log this event? if there's ever an issue with our `WHERE` clause we might not be tracking any spending metrics without ever knowing it

Reply:
Pretty sure we have a test to confirm that it's working.